### PR TITLE
perf: cut Apple Silicon latency via decode fast-path and MLX-native mel pipeline

### PR DESCRIPTION
**Pull request by moona3k**
## Context
This PR focuses on the highest-ROI latency bottlenecks in the Apple Silicon inference path, without changing model architecture or decoding behavior.

The objective is simple: reduce end-to-end transcription latency by removing avoidable overhead in the hottest paths.

### 1) Decode hot-path overhead reduction
Files:
- `mlx_qwen3_asr/model.py`
- `mlx_qwen3_asr/generate.py`
- `mlx_qwen3_asr/streaming.py`

Changes:
- Added optional `validate_input_ids` control to `Qwen3ASRModel.step()` / `step_many()`.
- Kept strict validation as the default public behavior (`validate_input_ids=True`).
- In internal autoregressive loops (generate/speculative/streaming), use `validate_input_ids=False` to skip per-token range scans that introduce host sync overhead.
- Added signature-aware fallback so test doubles / external call sites that do not accept the new kwarg remain compatible.

Safety:
- Unchecked decode path still enforces integer dtype validation.
- No model math, cache math, logits logic, or decoding policy changes.

### 2) Audio frontend conversion cleanup
File:
- `mlx_qwen3_asr/audio.py`

Changes:
- In `compute_features()`, removed unnecessary `MLX -> NumPy -> MLX` round-trip.
- Kept mel features in MLX end-to-end, including padding path.

Safety:
- Preserves output contract (shape, dtype, `feature_lens`, and `max_length` padding semantics).

## Benchmark Methodology
Command:
```bash
.venv/bin/python scripts/benchmark_asr.py tests/fixtures/test_speech.wav \
  --model Qwen/Qwen3-ASR-0.6B \
  --dtype float16 \
  --warmup-runs 2 \
  --runs 10
```

Lane:
- Audio: `tests/fixtures/test_speech.wav`
- Model: `Qwen/Qwen3-ASR-0.6B`
- Dtype: `float16`
- Machine context: same local Apple Silicon environment used for before/after.

## Results
Primary paired comparison (`runs=10`):

| Metric | Before | After | Delta |
|---|---:|---:|---:|
| Mean latency | 1.3168s | 1.0733s | **-18.5%** |
| RTF | 0.5198 | 0.4237 | **-18.5%** |
| Speedup | - | - | **1.23x** |

Additional confirmation run (`runs=12`) remained improved:
- Mean latency: `0.9724s`
- RTF: `0.3839`

## Validation
- `pytest -q`: **460 passed, 2 skipped**
- `scripts/quality_gate.py --mode fast`: **PASS**
- Targeted suites for touched areas also pass (`model`, `generate`, `audio`, `streaming`).

## Risk Assessment
- **Quality risk: low**
  - No decoder algorithm changes.
  - No weight/matmul/attention math changes.
- **API risk: low**
  - Strict validation remains default for external callers.
  - New fast path is opt-in and used internally in controlled hot loops.
- **Compatibility risk: low**
  - Internal callers use signature-aware fallback for models without the new kwarg.

## Commits
1. `perf(decode): add unchecked token-validation fast path for autoregressive steps`
2. `perf(audio): keep mel features in MLX and remove NumPy round-trip`

## Summary by CodeRabbit

* **New Features**
  * Added optional input validation control for model inference, allowing performance optimization in trusted scenarios.
  * Enhanced model compatibility detection to support different configurations and callable variants.

* **Refactor**
  * Optimized internal array handling and generation logic for improved efficiency.
  * Streamlined streaming and generation code paths with conditional validation support.

* **Tests**
  * Expanded test coverage for validation bypass scenarios and dtype enforcement.

### CODE REVIEW

File: mlx_qwen3_asr/generate.py [hunk 1]
```diff
@@ -278,6 +295,17 @@ def _sample(logits: mx.array, temperature: float) -> int:
...
 
+def _supports_validate_kwarg(callable_obj: object) -> bool:
+    """Return True when callable supports ``validate_input_ids`` kwarg."""
+    if callable_obj is None:
+        return False
+    try:
+        sig = inspect.signature(callable_obj)
+    except (TypeError, ValueError):
+        return False
+    return "validate_input_ids" in sig.parameters
```
[the remark this answers is not in the document]
Comment (moona3k) [reply mlx_qwen3_asr/generate.py]: Addressed. I extracted the duplicated helper into a shared utility (`mlx_qwen3_asr/runtime_utils.py`) as `supports_kwarg(...)`, and switched generation paths to import/use that single implementation. Commit: ee5d5f1.
